### ISSUE TITLE
[Feature Request] Android TV support

### ISSUE THREAD

**Issue by fp555** [NO-ob/LoliSnatcher_Droid#17]
Hello, do you have any plan to add full support for [Android TV?](https://developer.android.com/training/tv/start/start)
Being able to browse images from the sofa with a remote and voice search would be glorious.

I don't think any of the other Android *booru apps have full support either, so it could be a unique LoliSnatcher feature 😎

**Comment by NO-ob** [comment]
Do you know of a cheap android tv device?

**Comment by fp555** [comment]
I guess the [Xiaomi Mi Box (S)](https://www.amazon.com/dp/B07KLWGGYS) and the brand new [Chromecast with Google TV](https://store.google.com/us/product/chromecast_google_tv) would be the best choices.
Do not buy the super cheap chinese ones, they all have stock Android instead of Android TV except [Mecool boxes](https://www.mecoolonline.com/collections/mecool-atv).
There are many others, like the TiVo or the Jetstream, depending on where you live. The Amazon Fire Sticks are probably too different from vanilla ATV to recommend in this case. The Nvidia Shield is king here but it is NOT cheap!

Very happy you're considering doing this!

**Comment by NANI-SORE** [comment]
There is an android tv emulator option in android studio, so you probably don't need a real device to test this.
I tried it just now, but it crashed when I clicked on a virtual d-pad, probably needs some plugin imported to work and some logic to understand this type of controls.
I'm not sure voice search will really work when you constantly need to add delimeters/underscores/commas to tags, but still an interesting concept.

**Comment by fp555** [comment]
> I tried it just now, but the app crashed when I clicked on a virtual d-pad, probably needs some plugin imported to work and some logic to understand this type of controls.

I think by default android apps only work with a touchscreen and touch inputs (even a mouse won't always work, especially in games), you have to add specific code to handle input from a remote (dpad).

> I'm not sure voice search will really work when you constantly need to add delimeters/underscores/commas to tags, but still an interesting concept.

The thing with voice is that instead of writing with the onscreen keyboard and the remote dpad like a caveman, you could just say the tags and they will be converted to text. Or even better, you could say to the Assistant "search `stuff` on LoliSnatcher" in the same way you say "watch `show` on Netflix" to automatically open the app.

If your mom overhearing your fetish tags is not a concern (🤣) I think this would be peak comfy.

**Comment by NO-ob** [comment]
> I guess the [Xiaomi Mi Box (S)](https://www.amazon.com/dp/B07KLWGGYS) and the brand new [Chromecast with Google TV](https://store.google.com/us/product/chromecast_google_tv) would be the best choices.
> Do not buy the super cheap chinese ones, they all have stock Android instead of Android TV except [Mecool boxes](https://www.mecoolonline.com/collections/mecool-atv).
> There are many others, like the TiVo or the Jetstream, depending on where you live. The Amazon Fire Sticks are probably too different from vanilla ATV to recommend in this case. The Nvidia Shield is king here but it is NOT cheap!
> 
> Very happy you're considering doing this!

Thanks for the device list ill have a look into them


> There is an android tv emulator option in android studio, so you probably don't need a real device to test this.
> I tried it just now, but the app crashed when I clicked on a virtual d-pad, probably needs some plugin imported to work and some logic to understand this type of controls.
> I'm not sure voice search will really work when you constantly need to add delimeters/underscores/commas to tags, but still an interesting concept.

I bet the audio detection wont even get most tags correct aswell lel

**Comment by grravity** [comment]
I don't believe there is a way to introduce casting while remaining foss, so I would suggest two different builds (keep the one with casting only on the google play store).

**Comment by fp555** [comment]
I never mentioned casting, I just want a LoliSnatcher version that works flawlessly when installed on Android TV using a remote. A leanback layout, voice support and other bells and whistles would be the cherry on top, and I'd be totally fine with sticking to the bare minimum support at first.

**Comment by grravity** [comment]
I saw you mention Chromecast which is where that came from, my bad though
